src/gerbil/expander/root.ss

```diff
@@ -214,23 +214,28 @@ namespace: gx
       (match (string-split (symbol->string sys-type) #\-)
         (["linux" . rest] (not (null? rest)))
         (else #f)))
-    (def (bsd-variant? sys-type)
-      (def bsd ["openbsd" "netbsd" "freebsd" "darwin"])
-      (def sys-prefix
-        (list->string
-         (filter char-alphabetic?
-                 (string->list (symbol->string sys-type)))))
-      (not (null? (member sys-prefix bsd))))
+    (def (bsd-variant sys-type)
+      (let ((sys-prefix (list->string
+			  (filter char-alphabetic?
+				  (string->list
+				    (symbol->string sys-type))))))
+	(if (member sys-prefix ["openbsd" "netbsd" "freebsd" "darwin"])
+	  sys-prefix
+	  #f)))
     (core-bind-feature! 'gerbil #f 0 self)
     (core-bind-feature! (gerbil-system) #f 0 self)
     (match (system-type)
       ([sys-cpu sys-vendor sys-type]
        (core-bind-feature! sys-cpu #f 0 self)
        (core-bind-feature! sys-type #f 0 self)
-       (when (linux-variant? sys-type)
-         (core-bind-feature! 'linux #f 0 self))
-       (when (bsd-variant? sys-type)
-         (core-bind-feature! 'bsd #f 0 self)))
+       (cond
+	 ((linux-variant? sys-type)
+	  (core-bind-feature! (string->symbol "linux") #f 0 self))
+	 ((bsd-variant sys-type)
+	  => (lambda (sys-prefix)
+	       (core-bind-feature! (string->symbol "bsd") #f 0 self)
+	       (core-bind-feature!
```

Comment:
come think of this, isn't it already bound by the sys-type binding?

Reply:
referring to the exact system type, not the variant.

Reply:
`bsd-variant` returns "openbsd", "freebsd", ... , or `#f`. The sys-type binding for the BSDs in contrast also includes the version number of the kernel which is too fine grained. I needed a middleground to just make a distinction between the BSD branches. The idea is from alex-segura in his kqueue implementation.

Reply:
ok, this makes sense.